src/accelgyrotemp_ros_publisher.cpp

```diff
@@ -60,13 +63,47 @@ void AccelGyroTempRosPublisher::run()
         switch(operation_mode) {
         case DEVICE_CONTINUOUS_SAMPLING_MODE:
             success = false;
-            success = m_dataProvider->getData(m_message);
 
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_accelgyrotemp"), "Publishing acceleration data: '%f' '%f' '%f'",
-                        m_message.accel.x, m_message.accel.y, m_message.accel.z);
+            data_selection = m_node->get_parameter("data_selection").get_parameter_value().get<int32_t>();
 
-            if(success)
-                m_publisher->publish(m_message);
+            if(data_selection == PRINT_ACCELGYRO)
```

Comment:
let's change this if statement with a switch case, it will improve code readability.
please apply the same logic to the other publisher